RecoLocalTracker/SiStripZeroSuppression/src/SiStripAPVRestorer.cc

```diff
@@ -976,79 +942,280 @@ bool SiStripAPVRestorer::CheckBaseline(const std::vector<int16_t> &baseline) con
+        if ((abs(first_start_cluster_ADC - singleAPVdigi[strip+1])>first_gradient)&&(isPossible_wrong_minimum==true)) {
+          discontinuities.erase(itdiscontinuities);
+          discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip_first_gradient, ADC_start_point_cluster_pw));
+        }
+
+        if ((discontinuities.size()%2==0)&&(discontinuities.size()>1)){
+          itdiscontinuities=discontinuities.end();
+          --itdiscontinuities;
+          discontinuities.erase(itdiscontinuities);
+        }
+        discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip-1, previousStripADC));
+        isMax=false;
+        ADC_start_point_cluster_pw=0;
+        isPossible_wrong_minimum=false;
+        strip_first_gradient=0;
+        first_start_cluster_strip=0;
+        first_start_cluster_ADC=0;
+      }
     }
-    return apvFlagsBool_;	
-}
+  }
+
+  //----------THIRD PART reconstruction of the event without baseline-------//
 
+  if(!discontinuities.empty()){
+    if ((first_start_cluster_strip)==127-1) discontinuities.insert(discontinuities.end(), std::pair<int, int >(127, first_start_cluster_ADC));
+    if ((isMax==true)&&(isAuxiliary_Minimum==true)) discontinuities.insert(discontinuities.end(), std::pair<int, int >(auxiliary_end_cluster, first_start_cluster_ADC));
+  }
 
+  if (isFirstStrip==true){
+    itdiscontinuities=discontinuities.begin();
+    ++itdiscontinuities;
+    ++itdiscontinuities;
+    //int firstStrip= itdiscontinuities->first;
+    int firstADC= itdiscontinuities->second;
+    --itdiscontinuities;
+    itdiscontinuities->second=firstADC;
+    --itdiscontinuities;
+    itdiscontinuities->second=firstADC;
+    isFirstStrip=false;
+  }
 
+  if(!discontinuities.empty()){
+    itdiscontinuities=discontinuities.begin();
+    uint16_t firstStrip  = itdiscontinuities->first;
+    int16_t firstADC = itdiscontinuities->second;
+    ++itdiscontinuities;
+    uint16_t lastStrip  = itdiscontinuities->first;
+    int16_t secondADC = itdiscontinuities->second;
+    ++itdiscontinuities;
+
+    for(uint16_t strip=0; strip < 128; ++strip){
+      if (strip > lastStrip&&itdiscontinuities!=discontinuities.end()){
+        firstStrip  = itdiscontinuities->first;
+        firstADC = itdiscontinuities->second;
+        ++itdiscontinuities;
+        lastStrip  = itdiscontinuities->first;
+        secondADC = itdiscontinuities->second;
+        ++itdiscontinuities;
+      }
 
+      if ((firstStrip <= strip) && (strip <= lastStrip) && (0 < (singleAPVdigi[strip]- firstADC - ((secondADC-firstADC)/(lastStrip-firstStrip))*(strip-firstStrip)-gradient_threshold_))){
+        digis[(APVn-firstAPV)*128+strip]= singleAPVdigi[strip]- firstADC - (((secondADC-firstADC)/(lastStrip-firstStrip))*(strip-firstStrip));
+std::cout << "no baseline " << digis[(APVn-firstAPV)*128+strip] << std::endl;
```

Comment:
cout is not allowed in output of production code.
Please change to LogWarning if this condition is a problem happening once in ~100 events,
otherwise LogInfo or LogTrace